timelock-corruption-detection/src/test/java/com/palantir/timelock/corruption/detection/HistoryAnalyzerTest.java

```diff
@@ -18,178 +18,72 @@
-                localStateLogComponents.learnerLog(),
+                helper.getDefaultLocalServer().learnerLog(),
                 1,
                 PaxosSerializationTestUtils.createPaxosValueForRoundAndData(1, 1));
-        remoteStateLogComponents.forEach(server -> PaxosSerializationTestUtils.writePaxosValue(
-                server.learnerLog(), 1, PaxosSerializationTestUtils.createPaxosValueForRoundAndData(1, 5)));
-
-        List<CompletePaxosHistoryForNamespaceAndUseCase> historyForAll = paxosLogHistoryProvider.getHistory();
+        helper.getDefaultRemoteServerList()
+                .forEach(server -> PaxosSerializationTestUtils.writePaxosValue(
+                        server.learnerLog(), 1, PaxosSerializationTestUtils.createPaxosValueForRoundAndData(1, 5)));
+        List<CompletePaxosHistoryForNamespaceAndUseCase> historyForAll = helper.getHistory();
         assertThat(HistoryAnalyzer.divergedLearners(Iterables.getOnlyElement(historyForAll)))
                 .isEqualTo(CorruptionCheckViolation.DIVERGED_LEARNERS);
 
-        SetMultimap<CorruptionCheckViolation, NamespaceAndUseCase> setMultimap =
-                HistoryAnalyzer.corruptionHealthReportForHistory(historyForAll)
-                        .violatingStatusesToNamespaceAndUseCase();
-        assertThat(setMultimap.keySet()).containsExactly(CorruptionCheckViolation.DIVERGED_LEARNERS);
+        helper.assertViolationDetected(CorruptionCheckViolation.DIVERGED_LEARNERS);
     }
 
     @Test
     public void detectCorruptionIfLearnedValueIsNotAcceptedByQuorum() {
-        writeLogsOnServer(localStateLogComponents, 1, 10);
+        helper.writeLogsOnDefaultLocalServer(1, 10);
 
-        List<CompletePaxosHistoryForNamespaceAndUseCase> historyForAll = paxosLogHistoryProvider.getHistory();
+        List<CompletePaxosHistoryForNamespaceAndUseCase> historyForAll = helper.getHistory();
         assertThat(HistoryAnalyzer.divergedLearners(Iterables.getOnlyElement(historyForAll)))
                 .isEqualTo(CorruptionCheckViolation.NONE);
         assertThat(HistoryAnalyzer.learnedValueWithoutQuorum(Iterables.getOnlyElement(historyForAll)))
                 .isEqualTo(CorruptionCheckViolation.VALUE_LEARNED_WITHOUT_QUORUM);
 
-        SetMultimap<CorruptionCheckViolation, NamespaceAndUseCase> setMultimap =
-                HistoryAnalyzer.corruptionHealthReportForHistory(historyForAll)
-                        .violatingStatusesToNamespaceAndUseCase();
-        assertThat(setMultimap.keySet()).containsExactly(CorruptionCheckViolation.VALUE_LEARNED_WITHOUT_QUORUM);
+        helper.assertViolationDetected(CorruptionCheckViolation.VALUE_LEARNED_WITHOUT_QUORUM);
     }
 
     @Test
     public void detectCorruptionIfLearnedValueIsNotTheGreatestAcceptedValue() {
-        writeLogsOnServer(localStateLogComponents, 1, 5);
-        remoteStateLogComponents.forEach(server -> writeLogsOnServer(server, 1, 5));
+        helper.writeLogsOnDefaultLocalAndRemote(1, 10);
```

Comment:
As far as I can tell, every test uses the params `1, 10`. Perhaps you should mix this up a bit? Can these numbers be large (e.g. 10k)? Might help to catch any funny errors to have at least one test with large numbers - if that makes sense in prod.

Reply:
First part is done. For the second part I can add a (stress?) test in CorruptionDetectionIntegrationTest.